eng/versioning.targets

```diff
@@ -35,7 +35,14 @@
   <PropertyGroup Condition="'$(TargetsAnyOS)' == 'true' and !$(TargetFrameworks.Contains('$(TargetFramework)-Browser'))">
       <CrossPlatformAndHasNoBrowserTarget>true</CrossPlatformAndHasNoBrowserTarget>
   </PropertyGroup>
-  
+
+  <!-- Enables warnings for Android, iOS, and macCatalyst for all builds -->
```

Comment:
Should this be unified with the logic bellow and disable them for tests?

Reply:
I considered that, but thus far we've been OK having these platforms enabled for tests. I didn't want to loosen up any more than necessary, as that could add risk of dotnet/sdk#16488 introducing new warnings.